streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -2774,4 +2760,52 @@ object ZStreamSpec extends ZIOBaseSpec {
       testResult <- assertion(chunkCoordination)
     } yield testResult
 
+  private def inputStreamSuite = {
+    val bytes: List[Byte] = (-10 to 10).map(_.toByte).toList
+    val stream            = ZStream.fromIterable(bytes)
+
+    def testStream(assertion: java.io.InputStream => TestResult): ZIO[Any, Nothing, TestResult] =
+      ZIO
+        .foreach(List(1, 2, 5, 30)) { chunkSize =>
```

Comment:
Would it make sense to use a generator here?

Reply:
Added a generator and ended-up removing all my hand-written machinery.
Thanks!